workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/GetAppPage/components/ProductCard.tsx

```diff
@@ -3,131 +3,221 @@
  */
 
-import ClayBadge from '@clayui/badge';
 import {useEffect, useState} from 'react';
 
+import './ProductCard.scss';
+
+import ClaySticker from '@clayui/sticker';
+
+import emptyPictureIcon from '../../../assets/icons/avatar.svg';
 import {getProductById} from '../../../utils/api';
+import {getCustomFieldValue} from '../../../utils/customFieldUtil';
+import {
+	getThumbnailByProductAttachment,
+	getValueFromSpecifications,
+} from '../../../utils/util';
+import {LicenseType} from '../enums/licenseType';
+import {Price} from '../enums/price';
+import {SkuOptions} from '../enums/skuOptions';
 
 interface ProductCardProps {
 	productId: number | null;
 	selectedAccount?: Account;
 	setProductToForm: (product: Product) => void;
-	showAccount: Boolean;
 }
 
 const ProductCard = ({
 	productId,
 	selectedAccount,
 	setProductToForm,
-	showAccount,
 }: ProductCardProps) => {
-	const [product, setProduct] = useState<Product[]>([]);
+	const [product, setProduct] = useState<Product>();
+	const [hasTrial, setHasTrial] = useState<boolean>(false);
+	const [basePrice, setBasePrice] = useState<Number | undefined>(undefined);
+
+	const productHasTrialSKU = (skus: SKU[]) => {
+		skus.map((sku) => {
+			const licenseUsageType = sku.skuOptions.find(
+				(option) =>
+					option.key.toLowerCase() === 'dxp-license-usage-type'
+			);
+			if (
+				licenseUsageType &&
+				licenseUsageType.value.toLowerCase() ===
+					SkuOptions.TRIAL.toLowerCase()
+			) {
+				setHasTrial(true);
+
+				return;
+			}
+
+			return;
+		});
+	};
+
+	const getProductBasePrice = (product: Product) => {
+		product &&
+			product.skus.map((sku) => {
+				const licenseUsageType = sku.skuOptions.find(
+					(option) =>
+						option.key.toLowerCase() === 'dxp-license-usage-type'
+				);
+				if (
+					licenseUsageType &&
+					licenseUsageType.value.toLowerCase() ===
+						SkuOptions.STANDARD.toLowerCase()
+				) {
+					setBasePrice(sku.price);
+				}
+			});
+	};
 
 	useEffect(() => {
-		const getProdut = async () => {
-			// eslint-disable-next-line promise/catch-or-return
-			{
+		const fetchData = async () => {
+			const productResponse =
 				productId &&
-					getProductById({
-						nestedFields: 'skus',
-						productId,
-					}).then((item: Product) => {
-						setProduct([item]);
-					});
+				(await getProductById({
+					nestedFields: 'attachments,productSpecifications,skus',
+					productId,
+				}));
+
+			if (productResponse) {
+				setProduct(productResponse);
+				productHasTrialSKU(productResponse.skus);
+				setProductToForm(productResponse);
+				getProductBasePrice(productResponse);
 			}
 		};
-		getProdut();
-	}, [productId]);
 
-	setProductToForm(product[0]);
+		fetchData();
+	}, [productId, setProductToForm]);
+
+	const iconURL =
+		product &&
+		getThumbnailByProductAttachment(product.attachments)?.split('/o/');
+	const convertedIconURl = iconURL && `/o/${iconURL[1]}`;
```

Comment:
Let's say that iconURL is `false`,  convertedIconURl will be `false` also, is that intended behavior? 

```suggestion
	const convertedIconURl = iconURL ? `/o/${iconURL[1]}` :  '';
```

Reply:
```suggestion
	const convertedIconURL = iconURL && `/o/${iconURL[1]}`;
```